File: Backend/app/schemas/client/medical_record.py
from ..base import BaseSchema, BaseResponseSchema, add_padded_id
from typing import Optional


@add_padded_id()
class MedicalRecordResponse(BaseResponseSchema):
    """病历响应（患者端）"""
    appointment_id: int
    doctor_id: int
    diagnosis: str
    content: Optional[str] = None
    padded_id: Optional[str] = None
    doctor_name: Optional[str] = None

File: medical-client/src/api/medicalRecord.ts
import request from '../utils/request'
import type { PaginationResponse } from '../types/api'

export interface MedicalRecord {
  id: number
  appointment_id: number
  doctor_id: number
  diagnosis: string
  content?: string
  padded_id?: string
  doctor_name?: string
  created_at: string
  updated_at: string
}

export function listMyMedicalRecords(
  page = 1,
  perPage = 10
): Promise<PaginationResponse<MedicalRecord>> {
  return request.get('/medical-records', { params: { page, per_page: perPage } })
}
